### PR TITLE
Almost 100% on aea/helpers

### PR DESCRIPTION
**Pull request by Totoual**
## Fixes

Added tests for the aea/helpers.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Added tests
expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/connections/local/connection.py [hunk 1]
```diff
@@ -138,6 +140,17 @@ def register_service(self, public_key: str, service_description: Description):
         with self._lock:
             self.services[public_key].append(service_description)
 
+    def register_agent(self, public_key: str, agent_description: Description):
```
Comment (marcofavorito) [inline aea/connections/local/connection.py]: this is not used anywhere - shall we remove it?

File: aea/connections/local/connection.py [hunk 2]
```diff
@@ -178,9 +191,10 @@ def search_agents(self, public_key: str, search_id: int, query: Query) -> None:
         if query.model is None:
             result = list(set(self.services.keys()))
         else:
-            for agent_public_key, description in self.agents.items():
-                if query.model == description.data_model:
-                    result.append(agent_public_key)
+            for agent_public_key, descriptions in self.agents.items():
+                for description in descriptions:
+                    if query.model == description.data_model:
+                        result.append(agent_public_key)
```
Comment (marcofavorito) [inline aea/connections/local/connection.py]: notice, this was the old version. Here we should call the `check` method of the `query` object, which is now implemented.

File: aea/connections/local/connection.py (hunk 1 above)
Comment (Totoual) [reply aea/connections/local/connection.py]: It is pulled from the feature PR. I messed up with the PRs a little bit.

File: aea/connections/local/connection.py (hunk 2 above)
Comment (Totoual) [reply aea/connections/local/connection.py]: it is based on the service functionality.

Comment (marcofavorito) [reply aea/connections/local/connection.py]: Oh ok, I think we can remove also `search_agents` since it is not used (right?)

**Pull request merged**